src/cljs/main/broadfcui/page/method_repo/method/exporter.cljs

```diff
@@ -181,19 +182,38 @@
          :headers ajax/content-type=json
          :on-done (fn [{:keys [success? get-parsed-response]}]
                     (if success?
-                      (this :-export-loaded-config
-                            (merge (get-parsed-response) new-id {:rootEntityType dest-ret}))
+                      (this :-resolve-workspace (merge (get-parsed-response) new-id {:rootEntityType dest-ret}))
                       (utils/multi-swap! state (assoc :server-error (get-parsed-response false)) (dissoc :banner))))})))
+   :-resolve-workspace
+   (fn [{:keys [props state refs this]} config]
+     (if-let [workspace-id (:workspace-id props)]
+       (this :-export-loaded-config workspace-id config)
+       (let [{:keys [existing-workspace]
+              {:keys [project name description auth-domain]} :new-workspace}
```

Comment:
I don't like nested restructuring, maybe :keys :as and do the :new-workspace on another line?

Reply:
Phone won't let me edit, that was supposed to be destructuring